library/alloc/src/collections/btree/node.rs

```diff
@@ -910,7 +910,7 @@ impl<'a, K: 'a, V: 'a, NodeType> Handle<NodeRef<marker::Mut<'a>, K, V, NodeType>
     /// Inserts a new key/value pair between the key/value pairs to the right and left of
     /// this edge. This method assumes that there is enough space in the node for the new
     /// pair to fit.
-    fn leafy_insert_fit(&mut self, key: K, val: V) {
+    fn end_insert_fit(&mut self, key: K, val: V) {
```

Comment:
Why is this "end" insert fit? It doesn't seem to insert at the end, but rather at this particular edge?

Reply:
It ends the "insert_fit" process for a particular of the two node types. It doesn't absolutely need to be at the end, but it just is. There is no `begin_insert_fit`, matching `begin_split`, because there's no need. I can invent some other names, but this is always going to be confusing. Alternatives:

1. Write out the implementation for both leaf and internal functions, and stop trying to share a (big) part. Note I did not start sharing the 4 lines of code of `insert_fit`, I just tried to demystify it with #75058, and introduced the same principle on `split`. It seems much more worth while there, but mostly because of the `ptr::copy_nonoverlapping` calls that I don't think can be shortened without #44580.
2. Try to find convincing names and documentation for the "begin_" and "end_" functions as they exist now.
3. Write a single implementation with a lambda argument that treats edges or doesn't. Seems confusing too.
4. Write a single implementation for a generic edge. Tried as #77025.

Since 4. is too far, I'd rather retreat to 1 then try to clarify something in between.